library-model/library-model-generator/src/main/java/com/uber/nullaway/libmodel/LibraryModelGenerator.java

```diff
@@ -0,0 +1,276 @@
+
+    @Override
+    public void visit(MethodDeclaration md, Void arg) {
+      Optional<Node> parentClassNode = md.getParentNode();
+      String parentClassName = "";
+      if (parentClassNode.isPresent()) {
+        if (parentClassNode.get() instanceof ClassOrInterfaceDeclaration) {
+          parentClassName = ((ClassOrInterfaceDeclaration) parentClassNode.get()).getNameAsString();
+        } else if (parentClassNode.get() instanceof EnumDeclaration) {
+          parentClassName = ((EnumDeclaration) parentClassNode.get()).getNameAsString();
+        }
+      }
+      if (this.isNullMarked && hasNullableReturn(md)) {
+        methodRecords.put(
+            packageName
+                + "."
+                + parentClassName
+                + ":"
+                + getMethodReturnTypeString(md)
+                + " "
+                + md.getSignature().toString(),
+            MethodAnnotationsRecord.create(ImmutableSet.of("Nullable"), ImmutableMap.of()));
+      }
+      super.visit(md, null);
+    }
+
+    /**
+     * Determines if a MethodDeclaration can return null.
+     *
+     * @param md The MethodDeclaration instance.
+     * @return {@code true} if the method can return null, {@code false} otherwise.
+     */
+    private boolean hasNullableReturn(MethodDeclaration md) {
+      if (md.getType() instanceof ArrayType) {
+        /* For an Array return type the annotation is on the type when the Array instance is
+        Nullable(Object @Nullable []) and on the node when the elements inside are
+        Nullable(@Nullable Object []) */
+        for (AnnotationExpr annotation : md.getType().getAnnotations()) {
+          if (isAnnotationNullable(annotation)) {
+            return true;
+          }
+        }
+      } else {
+        for (AnnotationExpr annotation : md.getAnnotations()) {
+          if (isAnnotationNullable(annotation)) {
+            return true;
+          }
+        }
+      }
+      return false;
+    }
+
+    /**
+     * Takes a MethodDeclaration and returns the String value for the return type that will be
+     * written into the astubx file.
+     *
+     * @param md The MethodDeclaration instance.
+     * @return The return type string value to be written into the astubx file.
+     */
+    private String getMethodReturnTypeString(MethodDeclaration md) {
+      if (md.getType() instanceof ClassOrInterfaceType) {
+        return md.getType().getChildNodes().get(0).toString();
+      } else if (md.getType() instanceof ArrayType) {
+        return ARRAY_RETURN_TYPE_STRING;
+      } else {
+        return md.getType().toString();
+      }
+    }
+
+    private boolean isAnnotationNullable(AnnotationExpr annotation) {
+      // Since we only want to consider jspecify Nullable annotations.
+      return (annotation.getNameAsString().equalsIgnoreCase(NULLABLE)
+              && this.isJspecifyNullableImportPresent)
+          || annotation.getNameAsString().equalsIgnoreCase(JSPECIFY_NULLABLE_IMPORT);
+    }
```

Comment:
Do we care about star imports or other nonsense or is it fine here to only count either FQN or an import of the FQN?

Reply:
I think this logic is fine but we should add a comment that we don't consider star imports @akulk022